File: Interpretor2/src/Model/PrgState.java
package Model;

import java.io.BufferedReader;

public class PrgState {
	MyIStack<IStmt> exeStack;
	MyIDictionary<String,Integer> symTable;
	MyIList<Integer> out;
	MyIDictionary<Integer,Tuple<String,BufferedReader>> fileTable;
	MyIDictionary<Integer,Integer> heap;
	IStmt originalProgram;
	
	public PrgState(MyIStack<IStmt> e,	MyIDictionary<String,Integer> s,	MyIList<Integer> o,MyIDictionary<Integer,Tuple<String,BufferedReader>> t,MyIDictionary<Integer,Integer> h)
	{
		this.exeStack=e;
		this.symTable=s;
		this.out=o;
		this.fileTable=t;
		this.heap=h;
	}
	public MyIStack<IStmt> getExeStack()
	{
		return this.exeStack;
	}
	public MyIDictionary<String,Integer> getSymTable()
	{
		return this.symTable;
	}
	public MyIDictionary<Integer,Integer> getHeap()
	{
		return this.heap;
	}
	public MyIList<Integer> getOut()
	{
		return this.out;
	}
	public MyIDictionary<Integer,Tuple<String,BufferedReader>> getFileTable()
	{
		return this.fileTable;
	}
	@Override
	public String toString(){
		return "\nExeStack:\n"+this.exeStack.toString()+"\n-----------------------\n"+
				"\nSymTable:\n"+ this.symTable.toString()+"\n------------------------\n"+
				"Out:\n" + this.out.toString()+"\n------------------------\n"+
				"FileTable:\n"+this.fileTable.toString()+"\n------------------------\n"+
				"Heap:\n"+this.heap.toString()+"\n---------------------------------\n";
		//return this.exeStack.toString();
		}
	
}

File: MAP/Interpreter C#/Interpreter/PrgState.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace Interpreter
{
    namespace Model
    {
        class PrgState
        {
            MyIStack<IStmt> exeStack;
            MyIDictionary<String, int> symTable;
            MyIList<int> outList;
            MyIDictionary<int, Tuple<string,StreamReader>> fileTable;
            IStmt originalProgram;

            public PrgState(MyIStack<IStmt> stk, MyIDictionary<String, int> symtbl, MyIList<int> ot, MyIDictionary<int, Tuple<string, StreamReader>> fileTable)
            {
                this.exeStack = stk;
                this.symTable = symtbl;
                this.outList = ot;
                this.fileTable = fileTable;
            }
            public MyIStack<IStmt> getExeStack()
            {
                return this.exeStack;
            }
            public MyIDictionary<String, int> getSymTable()
            {
                return this.symTable;
            }
            public MyIList<int> getOut()
            {
                return this.outList;
            }
            public MyIDictionary<int, Tuple<string, StreamReader>> getFileTable()
            {
                return this.fileTable;
            }
            public override  String ToString()
            {
                return "\nExeStack:\n" + this.exeStack.ToString() + "\n-----------------------\n" +
                        "\nSymTable:\n" + this.symTable.ToString() + "\n------------------------\n" +
                        "Out:\n" + this.outList.ToString() + "\n------------------------\n" +
                        "FileTable:\n" + this.fileTable.ToString() + "\n-----------------------------\n";

            }


        }
    }
}
